### DISCUSSION TITLE
A question about realtime implementation ?

### DISCUSSION THREAD

**Discussion by regisx001** [pocketbase/pocketbase#2709] [Q&A]
Hello @ganigeorgiev 
first before all thank you for this awesome project . it a realy good project and it's inspired me to find myself in backend developement .

i want to ask you about the realtime part in the api . 
after reading the docs i see you implement it using SSE .
ok good for now ! 

i'm developing backend using django framework and it awesome .
but it's hard  to implement realtime API .
to do it there are 2 ways that i find : 
1 - websockets using `channels` , this way is too hard and very very advanced topics , and i have a lot of pain make it .
2 - SSE super easy to implement . but the probleme is the polling . aka send data in every 5 second or so ... .

but how did you implement the realtime part without polling , and when a record created the data sent to every subscriber to the record ? 

Thanks !!!!!

**Comment by ganigeorgiev** [comment]
Server-sent events (SSE) doesn't use polling. A SSE connection is a long lived HTTP connection and it behaves similarly to WebSockets. The main difference is that it is unidirectional (aka. only the server can send messages to the client). When the client want to send a message to the server, we send a separate http request.

For delivering/broadcasting messages, we use a simple fanout-like pattern to send a single event concurrently to all subscribers that are allowed to receive it.

More details about SSE and an example client-server implementation can be found in the [MDN SSE guide](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events).
